app/src/main/java/com/physphil/android/unitconverterultimate/util/Conversions.java

```diff
@@ -135,43 +137,63 @@ public void updateCurrencyConversions(Context context) {
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, DKK, R.string.dkk, Country.DKK));
+            addCurrencyUnitIfNotNull(units, new Unit(EUR, R.string.eur, 1.0, 1.0));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, HKD, R.string.hkd, Country.HKD));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, HUF, R.string.huf, Country.HUF));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, ISK, R.string.isk, Country.ISK));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, INR, R.string.inr, Country.INR));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, IDR, R.string.idr, Country.IDR));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, ILS, R.string.ils, Country.ILS));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, JPY, R.string.jpy, Country.JPY));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, KRW, R.string.krw, Country.KRW));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, MYR, R.string.myr, Country.MYR));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, MXN, R.string.mxn, Country.MXN));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, NZD, R.string.nzd, Country.NZD));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, NOK, R.string.nok, Country.NOK));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, PHP, R.string.php, Country.PHP));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, PLN, R.string.pln, Country.PLN));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, RON, R.string.ron, Country.RON));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, SGD, R.string.sgd, Country.SGD));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, ZAR, R.string.zar, Country.ZAR));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, SEK, R.string.sek, Country.SEK));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, CHF, R.string.chf, Country.CHF));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, THB, R.string.thb, Country.THB));
+            addCurrencyUnitIfNotNull(units, getCurrencyUnit(map, LIRA, R.string.lira, Country.TRY));
         }
 
         addConversion(Conversion.CURRENCY, new Conversion(Conversion.CURRENCY, R.string.currency, units));
     }
 
+    @Nullable
+    private Unit getCurrencyUnit(Map<Country, Double> currencyMap, int id, int labelResource, Country country) {
+        try {
+            return new Unit(
+                /* id */ id,
+                /* labelResource */ labelResource,
+                /* conversionToBase */ 1 / currencyMap.get(country),
+                /* conversionFromBase */ currencyMap.get(country)
+            );
+        } catch (NullPointerException ex) {
+            // Currency not present
+            return null;
+        }
```

Comment:
`NullPointerException` can be thrown at several places, catching and ignoring it can hide programming errors. Instead, I’d add an explicit null check:
```java
Double rate = currencyMap.get(country);
return rate != null ? new Unit(id, labelResource, 1 / rate, rate) : null;
```